doc: add 'Release Phases' documentation

BethGriggs opened the pull request:
Please let me know if you think this should be moved elsewhere.

BridgeAR commented on README.md:
```diff
@@ -20,6 +20,14 @@ Dates are subject to change.
 
+### Release Phases
+
+There are three phases that a Node.js release can be in: 'Current', 'Active Long-term Support', and 'Maintenance'. Odd-numbered release lines are not promoted to LTS - they will not go through the 'Active LTS' or 'Maintenance' phases. 
```
Nit: it would be good to keep the 80 character limit as in the rest of the file.

MylesBorins commented:
Overall this LGTM but I am noticing a bit of a duplication of content between this new section and the [release plan section](https://github.com/nodejs/Release#release-plan).

Should we perhaps remove duplicated content?

BethGriggs commented:
@MylesBorins I'll try and morph the content into the release plan

BethGriggs commented:
I've updated this quite significantly to try and reduce duplication. The main changes are:

- Added release phases section. 
  - Hopefully would clarify  https://github.com/nodejs/Release/issues/430

- Moved and shortened the "Changes in an LTS-covered major version are limited to" list to the following text: 
> New features, bug fixes, and updates that have been audited by the LTS team and have been determined to be appropriate and stable for the release line. Changes required for critical security and bug fixes may lead to *semver-major* changes landing within an LTS stream, such situations will be rare and will land as *semver-minor*.

The above text is far less specific than what we had before, and defaults to "at the discretion of the release/LTS teams".  Please let me know your thoughts on removing the list of specifics changes we land such as npm updates, etc. 

- Added "supported" release lines term definition. - #359

richardlau commented on README.md:
```diff
@@ -15,6 +15,27 @@ Dates are subject to change.
+### Release Phases
+
+There are three phases that a Node.js release can be in: 'Current', 'Active
+Long Term Support', and 'Maintenance'. Odd-numbered release lines are not
```
```suggestion
Long Term Support (LTS)', and 'Maintenance'. Odd-numbered release lines are not
```

BethGriggs commented:
Addressed recent comments from @richardlau, @MylesBorins - I think this is good to go now

MylesBorins commented on README.md:
```diff
@@ -15,6 +15,29 @@ Dates are subject to change.
+
+There are three phases that a Node.js release can be in: 'Current', 'Active
+Long Term Support (LTS)', and 'Maintenance'. Odd-numbered release lines are not
+promoted to LTS - they will not go through the 'Active LTS' or 'Maintenance'
```
I thought that odd number releases went through a short maintenance period? is that no longer true?

BethGriggs replied:
It is true according to the `schedule.json`- I think we've been purposely vague about it in the past. We can change that - maybe something like "release lines may receive critical fixes shortly after going EOL at the discretion of the Release team"? But I am not sure which is the best approach personally.

MylesBorins commented on README.md:
```diff
@@ -15,6 +15,29 @@ Dates are subject to change.
+ the new feature supports migration to later release lines.
+
+Changes required for critical security and bug fixes may lead to *semver-major*
+changes landing within an LTS stream, such situations will be rare and will
```
this could happen in LTS or Current afaik
```suggestion
changes landing within a release stream, such situations will be rare and will
```

MylesBorins commented on README.md:
```diff
@@ -70,72 +93,30 @@ Working Group.
+In coordination with a new *odd-numbered* major release, the previous
+*even-numbered* major version will transition to Long Term Support. The
+transition to Long Term Support will happen in a *semver-minor* release and can
+ happen either before or after the new major version is released.
```
We may want to open a follow up issue to track this, if I recall this is the exact thing that caused the "two current releases" thing.

As we will also be transitioning Active LTS releases to maintenance at the same time now we may want to have slightly more robust / clear policy here.

To be clear, not blocking... this should be a follow up.

BethGriggs replied:
Agreed - I think https://github.com/nodejs/Release/issues/495 had some of the conversation tied to it
